### PR TITLE
feat: Improve OTel wrapper ergonomics

### PR DESCRIPTION
**Pull request by anticorrelator**
- The high-level OTel wrappers will now print a summary of the TracerProvider that has been configured
- Defaults Phoenix endpoint to: `http://localhost:6006`
- Better handling of endpoints set via environment variables, defaults to using GRPC transport for spans
- Fixes small bugs

### CODE REVIEW

File: packages/phoenix-otel/src/phoenix/otel/otel.py [hunk 1]
```diff
@@ -78,80 +94,107 @@ def __init__(self, *args: Any, endpoint: Optional[str] = None, **kwargs: Any):
...
+                else:
+                    processor_name = "Multiple Span Processors"
+                    endpoint = "Multiple Span Exporters"
+                    transport = "Multiple Span Exporters"
+                    headers = "Multiple Span Exporters"
+
+        if os.name == "nt":
+            details_header = "OpenTelemetry Tracing Details"
+        else:
+            details_header = "🔭 OpenTelemetry Tracing Details 🔭"
```
Comment (mikeldking) [inline packages/phoenix-otel/src/phoenix/otel/otel.py:143]: should the guard be just for windows?

File: packages/phoenix-otel/src/phoenix/otel/otel.py [hunk 2]
```diff
@@ -22,17 +23,19 @@
...
+
 
 def register(
     *,
     endpoint: Optional[str] = None,
     project_name: Optional[str] = None,
     batch: bool = False,
-    set_global=True,
-    headers=None,
+    set_global_tracer: bool = True,
```
Comment (mikeldking) [inline packages/phoenix-otel/src/phoenix/otel/otel.py:34]:
```suggestion
    set_global_tracer_provider: bool = True,
```

I guess more explicit maybe

File: packages/phoenix-otel/src/phoenix/otel/otel.py [hunk 3]
```diff
@@ -22,17 +23,19 @@
...
+    headers: Optional[Dict[str, str]] = None,
```
Comment (mikeldking) [inline packages/phoenix-otel/src/phoenix/otel/otel.py:35]: Should verbose be plumbed through here too?

File: packages/phoenix-otel/src/phoenix/otel/otel.py (hunk 1 above)
Comment (anticorrelator) [reply packages/phoenix-otel/src/phoenix/otel/otel.py:143]: The current guard is only for Windows, are we worried about anything else? (cc @RogerHYang)

File: packages/phoenix-otel/src/phoenix/otel/otel.py [hunk 4]
```diff
@@ -192,3 +237,45 @@ def _maybe_grpc_endpoint(parsed_endpoint: ParseResult) -> bool:
...
+def _exporter_transport(exporter: SpanExporter) -> str:
+    if isinstance(exporter, _HTTPSpanExporter):
+        return "HTTP"
+    if isinstance(exporter, _GRPCSpanExporter):
+        return "gRPC"
+    else:
+        return exporter.__class__.__name__
+
+
+def _normalize_headers(headers: Union[List[Tuple[str, str]], Dict[str, str]]) -> Dict[str, str]:
```
Comment (mikeldking) [inline packages/phoenix-otel/src/phoenix/otel/otel.py:251]: Normalize is probably not the right word. It's just for printing so maybe something like https://github.com/Arize-ai/phoenix/blob/ce00d50930589f97d738f05f39d3d7b3c145e967/src/phoenix/db/engines.py#L32

**Review by mikeldking — APPROVED** [review]
LGTM - will try it out tomorrow.

**Pull request merged**